src/schema/v2/homeView/sections.ts

```diff
@@ -14,15 +14,19 @@ import {
 } from "./artistResolvers"
 import { HeroUnitsResolver } from "./heroUnitsResolver"
 
-type MaybeResolved<T> = T | ((context: ResolverContext) => Promise<T>)
+type MaybeResolved<T> =
+  | T
+  | ((context: ResolverContext, args: any) => Promise<T>)
 
 export type HomeViewSection = {
   id: string
   type: string
   component?: {
     title?: MaybeResolved<string>
+    type?: MaybeResolved<string>
```

Comment:
**thought (non-blocking):** In my spike I had this as just `string` instead of `MaybeResolved<string>`. 

Specifying preferred components seems to me a concern of MP's new home view schema, rather than something which is likely to be fetched from _upstream_.  

That's just my best guess, so I'm not insisting — but I might be in favor of simplifying this to just `string`. I admit knowing this distinction ahead of time when introducing a new field might be challenging. (Seems fine for the other fields introduced here.)

Reply:
Totally makes sense, I did it on autopilot - it's very unlikely `type` will ever be not a regular hardcoded string

Reply:
I will address in follow-up PR if you don't mind just for my convenience